### PR TITLE
Fix_max_query_size_for_kql_compound_operator:

### PR DESCRIPTION
**Pull request by kashwy**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Fix the issue of `max_query_size` for KQL compound operator like mv-expand. Related to #59626.

### CODE REVIEW

**Comment by kashwy** [comment]
Hi @antaljanosbenjamin , can you review  this PR,  this fixed another use case for kql operators like mv-expand (for https://github.com/ClickHouse/ClickHouse/pull/59626)

Thanks

File: tests/queries/0_stateless/02366_kql_mvexpand.sql [hunk 1]
```diff
@@ -33,3 +33,7 @@ print '-- mv_expand_test_table | mv-expand with_itemindex=index c,d to typeof(bo
 mv_expand_test_table | mv-expand with_itemindex=index c,d to typeof(bool);
 print '-- mv_expand_test_table | mv-expand c to typeof(bool) --';
 mv_expand_test_table | mv-expand c to typeof(bool);
+SET max_query_size = 28;
+SET dialect='kusto';
+mv_expand_test_table | mv-expand c, d; -- { serverError 62 }
```
Comment (antaljanosbenjamin) [inline tests/queries/0_stateless/02366_kql_mvexpand.sql:38]:
```suggestion
mv_expand_test_table | mv-expand c, d; -- { serverError SYNTAX_ERROR }
```

**Review by antaljanosbenjamin — APPROVED** [review]
The changes looks great, only a tiny change for better readability.

**Review by antaljanosbenjamin — APPROVED** [review]

**Comment by antaljanosbenjamin** [comment]
Tests:
- `test_storage_azure_blob_storage/test.py::test_union_schema_inference_mode` https://github.com/ClickHouse/ClickHouse/issues/60347
- `02908_many_requests_to_system_replicas` https://github.com/ClickHouse/ClickHouse/issues/57443
- `AST fuzzer` is a tricky one. I am 99% sure the issue coming from KQL, but I couldn't reproduce it. Based on [CI DB](https://play.clickhouse.com/play?user=play#U0VMRUNUIGNoZWNrX3N0YXJ0X3RpbWUsIGNoZWNrX25hbWUsIHRlc3RfbmFtZSwgdGVzdF9zdGF0dXMsIGNoZWNrX3N0YXR1cywgcmVwb3J0X3VybApGUk9NIGNoZWNrcwpXSEVSRSAxCiAgICAtLSBBTkQgcHVsbF9yZXF1ZXN0X251bWJlciA9IDAKICAgIEFORCB0ZXN0X3N0YXR1cyAhPSAnU0tJUFBFRCcKICAgIEFORCB0ZXN0X3N0YXR1cyAhPSAnT0snCiAgICBBTkQgY2hlY2tfc3RhdHVzICE9ICdzdWNjZXNzJwogICAgQU5EIHRlc3RfbmFtZSBpbGlrZSAnJUxvZ2ljYWwgZXJyb3I6IFwnc3RkOjpleGNlcHRpb24uIENvZGU6IDEwMDEsIHR5cGU6IHN0ZDo6bGVuZ3RoX2Vycm9yLCBlLndoYXQoKSA9IGJhc2ljX3N0cmluZyUnCk9SREVSIEJZIGNoZWNrX3N0YXJ0X3RpbWUgZGVzYywgY2hlY2tfbmFtZSwgdGVzdF9uYW1l) the same issue pops up on other PRs too. So my hypothesis is we still have some issues in KQL parsers, but this PR definitely fixes one and don't introduce another one. Therefore I will merge this PR.

**Pull request merged**